test/test_tests_scan.py

```diff
@@ -108,147 +117,101 @@ def setUp(self):
+            f" && cd make-checkout-workdir && TEST_OS=fedora BASE_BRANCH=stable-1.0"
+            " COCKPIT_BOTS_REF=main TEST_SCENARIO=nightly ../tests-invoke --pull-number"
+            f" {self.pull_number} --revision {self.revision} --repo {self.repo}\"\n")
+
+        self.expected_human_output = (
+            f"pull-{self.pull_number}      {self.context}            {self.revision}"
+            f"     5.99999  ({self.repo}) [bots@main]   {{stable-1.0}}\n")
+
     def tearDown(self):
         self.server.kill()
         shutil.rmtree(self.temp)
 
-    def run_tests_scan(self, args):
-        script = os.path.join(BOTS_DIR, "tests-scan")
-        proc = subprocess.Popen([script, *args], stdout=subprocess.PIPE, universal_newlines=True)
-        output, stderr = proc.communicate()
-        return proc, output, stderr
-
-    def expected_command(self):
-        return (f"./s3-streamer --repo {self.repo} --test-name pull-{self.pull_number}-\\d+-\\d+"
-                f" --github-context {self.context} --revision {self.revision} -- /bin/sh -c"
-                f" \"PRIORITY=0005 ./make-checkout --verbose --repo={self.repo} --rebase=stable-1.0 {self.revision}"
-                f" && cd make-checkout-workdir && TEST_OS=fedora BASE_BRANCH=stable-1.0"
-                " COCKPIT_BOTS_REF=main TEST_SCENARIO=nightly ../tests-invoke --pull-number"
-                f" {self.pull_number} --revision {self.revision} --repo {self.repo}\"")
-
-    def expected_human_output(self, pull_number=None):
-        if pull_number is None:
-            pull_number = self.pull_number
-        return (f"pull-{pull_number}      {self.context}            {self.revision}"
-                f"     5.99999  ({self.repo}) [bots@main]   {{stable-1.0}}")
+    @unittest.mock.patch("sys.stderr", new_callable=io.StringIO)
+    @unittest.mock.patch("sys.stdout", new_callable=io.StringIO)
+    # fake the time so that we get predictable test names
+    @unittest.mock.patch("time.strftime", return_value="20240102-030405")
+    def run_tests_scan(self, args, _mock_strftime, mock_stdout, mock_stderr, repo=None):
+        with unittest.mock.patch("sys.argv", ["tests-scan", "--repo", repo or self.repo, *args]):
+            try:
+                self.tests_scan_module.main()
+                code = 0
+            except SystemExit as e:
+                code = e.code
 
-    def test_pull_number(self):
-        args = ["--dry", "--repo", self.repo, "--pull-number", self.pull_number,
-                "--context", self.context]
-        proc, output, stderr = self.run_tests_scan(args)
+        return code, mock_stdout.getvalue(), mock_stderr.getvalue()
 
-        self.assertEqual(proc.returncode, 0)
-        expected_output = self.expected_command()
-        self.assertRegex(output, expected_output)
-        self.assertIsNone(stderr)
+    def run_success(self, args, expected_output, repo=None):
```

Comment:
(volume low, but gradually increasing) I would appreciate it if we type-annotated new code...